src/target/rp.c

```diff
@@ -387,6 +388,17 @@ static bool rp_flash_erase(target_flash_s *f, target_addr_t addr, size_t len)
 {
 	DEBUG_INFO("Erase addr 0x%08" PRIx32 " len 0x%" PRIx32 "\n", addr, (uint32_t)len);
 	target *t = f->t;
+
+	/* Update our assumptions with the SFDP params */
+	spi_parameters_s spi_parameters;
+	rp_flash_s *flash = (rp_flash_s *)f;
+	if (!(flash->table_has_been_read) && sfdp_read_parameters(t, &spi_parameters, rp_spi_read_sfdp)) {
```

Comment:
The brackets here around table_has_been_read are not required and suppress a GCC warning we want enabled. Please remove them

Reply:
Fixed!